### PR TITLE
Start threads at the same time

### PR DESCRIPTION
**Pull request by charmoniumQ**

### CODE REVIEW

**Comment by mhuzai** [comment]
Everything prior to the pause functionality has been tested thoroughly and looks good. Do you wish to keep the pause functionality or was it only in case the previous changes didn't work out?

**Comment by charmoniumQ** [comment]
> Everything prior to the pause functionality has been tested thoroughly and looks good. Do you wish to keep the pause functionality or was it only in case the previous changes didn't work out?

If it is up to me, I recommend including the "separated start method" but not the full pause functionality, which is kind of a hack. I can easily make this change if you agree.

**Comment by mhuzai** [comment]
> > Everything prior to the pause functionality has been tested thoroughly and looks good. Do you wish to keep the pause functionality or was it only in case the previous changes didn't work out?
> 
> If it is up to me, I recommend including the "separated start method" but not the full pause functionality, which is kind of a hack. I can easily make this change if you agree.

Yup, I agree with you.

And in that case, this is good to go once you make the change and update to latest master.

File: common/gl_util/lib/stb_image.h [hunk 1]
```diff
@@ -4120,7 +4120,7 @@ static int stbi__zhuffman_decode_slowpath(stbi__zbuf *a, stbi__zhuffman *z)
    if (s >= 16) return -1; // invalid code!
    // code size is s, so:
    b = (k >> (16-s)) - z->firstcode[s] + z->firstsymbol[s];
-   if (b >= (int) sizeof (z->size)) return -1; // some data was corrupt somewhere!
+   if (size_t(b) >= sizeof (z->size)) return -1; // some data was corrupt somewhere!
```
Comment (mhuzai) [inline common/gl_util/lib/stb_image.h:4123]: Please revert.

File: common/plugin.hpp [hunk 2]
```diff
@@ -22,12 +22,17 @@ namespace ILLIXR {
...
 		/**
-		 * @brief A method which Spindle calls when it starts the component.
-		 * 
+		 * @brief A method which Spindle called after the constructor.
+		 *
 		 * This is necessary for actions which have to be started after constructions, such as
 		 * threads. These cannot be started in the constructor because virtual methods don't work in
 		 * consturctors.
 		 */
+		virtual void setup() { }
```
Comment (mhuzai) [inline common/plugin.hpp:31]: Is this still required? Given that there is no pause functionality?

File: common/gl_util/lib/stb_image.h (hunk 1 above)
Comment (charmoniumQ) [reply common/gl_util/lib/stb_image.h:4123]: I think this is better. Extending an `int` to a `size_t` is safer than `size_t` to `int`.

**Review by apache8080 — DISMISSED** [review]

**Review by apache8080 — APPROVED** [review]

**Pull request merged**